Add options for from-branch and to-branch

JustinKuli opened the pull request:
These were previously hard-coded to "master", but it was straightforward to enable them as an option and it seems to work in my tests. (Speaking of tests, I wasn't sure how to test different branches in a unit test...)

I'm hoping that `EnvLocale` will help prevent too many extra parameters in the Promote function. Otherwise, this change would add 2 more, and specifying environments (#92) might add even more.

I may have gone a little bit over-zealous refactoring when putting in `EnvLocale`, so please let me know if I should scale that back.

mnuttall commented:
Hi @JustinKuli please can you rebase this and get a clean Travis build? 

`EnvLocale` isn't very self-explanatory. 

```go
type EnvLocale struct {
 	Path   string // URL or local path
 	Branch string
 }
```

I can see what's in it but like you I'm wondering what's the best name for this.

JustinKuli commented:
Rebased, but there were some odd conflicts (as expected). I'll need to review the diffs again, but I think this is ready for comments at least.

JustinKuli commented:
@mnutall, rebased and fixed up again. I agree that `EnvLocale` is not ideal. I got there from trying to shorten `EnvironmentLocation`, since it describes where the environment is (git repo, branch). I wonder if something like `GitCoordinates` would be better?

JustinKuli commented:
@mnuttall I'd like to build off of this for the work in #92. Is this struct clear enough?
```
type EnvLocation struct {
	RepoPath string // URL or local path
	Branch   string
}
```
If the "local" case gets handled separately in the future, we could rename RepoPath to something which is more clear that it's just a URL, but I'm not sure what other changes we could make right now.

mnuttall commented on pkg/cmd/promote.go:
```diff
@@ -140,7 +156,18 @@ func promoteAction(c *cobra.Command, args []string) error {
+		Branch:   toBranch,
+	}
+
+	sm := promotion.New(cacheDir, author, promotion.WithDebug(debug), promotion.WithInsecureSkipVerify(insecureSkipVerify), promotion.WithRepoType(repoType))
```
Much clearer : )
